lib/request/flushGetRequest.js

```diff
@@ -21,42 +21,56 @@ module.exports = function flushGetRequest(request, listOfPaths, callback) {
-    var listIdx = 0, listLen = listKeys.length;
-    for (; listIdx < listLen; ++listIdx) {
-        var paths = listOfPaths[listIdx];
-        for (var j = 0, pathLen = paths.length; j < pathLen; ++j) {
-            var pathSet = paths[j];
-            var len = pathSet.length;
+    var model = request.requestQueue.model;
+    if (model._enablePathCollapse || model._enableRequestDeduplication) {
+        // Note on the if-condition: request deduplication uses request._pathMap,
+        // so we need to populate that field if the feature is enabled.
 
-            if (!pathMap[len]) {
-                pathMap[len] = [pathSet];
-            } else {
-                var pathSetsByLength = pathMap[len];
-                pathSetsByLength[pathSetsByLength.length] = pathSet;
+        // TODO: Move this to the collapse algorithm,
+        // TODO: we should have a collapse that returns the paths and
+        // TODO: the trees.
+
+        // Take all the paths and add them to the pathMap by length.
+        // Since its a list of paths
+        var pathMap = request._pathMap;
+        var listIdx = 0,
+            listLen = listOfPaths.length;
+        for (; listIdx < listLen; ++listIdx) {
+            var paths = listOfPaths[listIdx];
+            for (var j = 0, pathLen = paths.length; j < pathLen; ++j) {
+                var pathSet = paths[j];
+                var len = pathSet.length;
+
+                if (!pathMap[len]) {
+                    pathMap[len] = [pathSet];
+                } else {
+                    var pathSetsByLength = pathMap[len];
+                    pathSetsByLength[pathSetsByLength.length] = pathSet;
+                }
             }
         }
-    }
 
-    // now that we have them all by length, convert each to a tree.
-    var pathMapKeys = Object.keys(pathMap);
-    var pathMapIdx = 0, pathMapLen = pathMapKeys.length;
-    for (; pathMapIdx < pathMapLen; ++pathMapIdx) {
-        var pathMapKey = pathMapKeys[pathMapIdx];
-        pathMap[pathMapKey] = toTree(pathMap[pathMapKey]);
+        // now that we have them all by length, convert each to a tree.
+        var pathMapKeys = Object.keys(pathMap);
+        var pathMapIdx = 0,
+            pathMapLen = pathMapKeys.length;
+        for (; pathMapIdx < pathMapLen; ++pathMapIdx) {
+            var pathMapKey = pathMapKeys[pathMapIdx];
+            pathMap[pathMapKey] = toTree(pathMap[pathMapKey]);
+        }
     }
 
-    // Take the pathMapTree and create the collapsed paths and send those
-    // off to the server.
-    var collapsedPaths = request._collasped = toPaths(pathMap);
-    var jsonGraphData;
+    if (model._enablePathCollapse) {
+        // Take the pathMapTree and create the collapsed paths and send those
+        // off to the server.
+        requestPaths = toPaths(request._pathMap);
+    } else if (listOfPaths.length === 1) {
+        requestPaths = listOfPaths[0];
+    } else {
+        requestPaths = Array.prototype.concat.apply([], listOfPaths);
```

Comment:
what's wrong with sending `listOfPaths` itself to `model._source.get()`, instead of sending a copy?

Reply:
This line performs a shallow flatten of `listOfPaths`:

`[[["todos", [0, 1], "description"]], [["items", [2, 3], "title"]]]`
=>
`[["todos", [0, 1], "description"], ["items", [2, 3], "title"]]`

Despite the name, `listOfPaths` is an Array of Arrays of path sets, while `DataSource#get` expected an Array of path sets.